src/pal/src/init/pal.cpp

```diff
@@ -171,6 +174,53 @@ PAL_InitializeDLL()
 
 /*++
 Function:
+  InitializeDefaultStackSize
+
+Abstract:
+  Initializes the default stack size. 
+
+--*/
+__attribute__((noinline,optnone))
+void
+InitializeDefaultStackSize()
+{
+    char* defaultStackSizeStr = getenv("COMPlus_DefaultStackSize");
+    if (defaultStackSizeStr != NULL)
+    {
+        errno = 0;
+        // Like all numeric values specific by the COMPlus_xxx variables, it is a 
+        // hexadecimal string without any prefix.
+        long int size = strtol(defaultStackSizeStr, NULL, 16);
+
+        if (errno == 0)
+        {
+            g_defaultStackSize = max(size, PTHREAD_STACK_MIN);
+        }
+    }
+
+#ifndef __GLIBC__
+    // This fixes a problem on MUSL where the initial stack size reported by the
+    // pthread_attr_getstack is about 128kB, but this limit is not fixed and
+    // the stack can grow dynamically. The problem is that it makes the 
+    // functions ReflectionInvocation::[Try]EnsureSufficientExecutionStack 
+    // to fail for real life scenarios like e.g. compilation of corefx.
+    // Since there is no real fixed limit for the stack, the code below
+    // ensures moving the stack limit to a value that makes reasonable
+    // real life scenarios work.
+    if (g_defaultStackSize == 0)
+    {
+        // Set the default minimum stack size for MUSL to the same value as we
+        // use on Windows.
+        g_defaultStackSize = 1536 * 1024;
+    }
+
+    uint8_t *s = (uint8_t *)_alloca(g_defaultStackSize);
+    s[0] = 0;
```

Comment:
It maybe nice to move the small part that does the alloca moved into separate method and mark just that - more readable, and probably smaller too.